extensions/keycloak-authorization/deployment/src/test/java/io/quarkus/keycloak/pep/test/KeycloakAuthorizationOfflineEnforcerTest.java

```diff
@@ -0,0 +1,101 @@
+import org.keycloak.representations.AccessTokenResponse;
+
+import io.quarkus.keycloak.pep.runtime.PolicyEnforcerResolver;
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+
+public class KeycloakAuthorizationOfflineEnforcerTest {
+
+    @RegisterExtension
+    final static QuarkusUnitTest app = new QuarkusUnitTest()
+            .withApplicationRoot(jar -> jar
+                    .addClasses(AdminResource.class, PublicResource.class, UserResource.class)
+                    .addAsResource("offline-enforcer-application.properties", "application.properties")
+                    .addAsResource("quarkus-realm.json"));
+
+    @Inject
+    PolicyEnforcerResolver resolver;
+
+    @Inject
+    @ConfigProperty(name = "keycloak.url")
+    String keycloakUrl;
+
+    @Test
+    public void testNoPathMatcherRemoteCalls() {
+
+        // first test Keycloak policy enforcer works
+        testPublicResource();
+        testAccessUserResource();
+        testAccessAdminResource();
+
+        // as we can't test whether there were remote calls without extensive logging
+        // here we test whether expected decisions are made
+        testIsPathEnforcementDisabled();
+    }
+
+    private void testIsPathEnforcementDisabled() {
+        var defaultEnforcer = resolver.getPolicyEnforcer(null);
+        assertTrue(defaultEnforcer.isPathEnforcementDisabled("/api/public/serve"));
+        assertTrue(defaultEnforcer.isPathEnforcementDisabled("/api/public"));
+        assertFalse(defaultEnforcer.isPathEnforcementDisabled("/api"));
+        assertFalse(defaultEnforcer.isPathEnforcementDisabled("/api/users"));
+        assertFalse(defaultEnforcer.isPathEnforcementDisabled("/api/users/me"));
+    }
+
+    private void testAccessUserResource() {
+        RestAssured.given().auth().oauth2(getAccessToken("alice"))
+                .when().get("/api/users/me")
+                .then()
+                .statusCode(200);
+        RestAssured.given().auth().oauth2(getAccessToken("jdoe"))
+                .when().get("/api/users/me")
+                .then()
+                .statusCode(200);
+    }
+
+    private void testAccessAdminResource() {
+        RestAssured.given().auth().oauth2(getAccessToken("alice"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(403);
+        RestAssured.given().auth().oauth2(getAccessToken("jdoe"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(403);
+        RestAssured.given().auth().oauth2(getAccessToken("admin"))
+                .when().get("/api/admin")
+                .then()
+                .statusCode(200);
+    }
+
+    private void testPublicResource() {
+        RestAssured.given()
+                .when().get("/api/public/serve")
+                .then()
+                .statusCode(200);
+    }
+
+    private String getAccessToken(String userName) {
```

Comment:
@michalvavrik This is fine, but it can be simplified if instead of injecting `keycloak.url` you create a new `KeycloakTestClient` instance as a test class field and then do `testClient.accessToken(...)`, only FYI, np if you prefer to keep this code.